app/scripts/controllers/mmi-controller.js

```diff
@@ -31,6 +31,7 @@ export default class MMIController extends EventEmitter {
     this.opts = opts;
     this.mmiConfigurationController = opts.mmiConfigurationController;
     this.keyringController = opts.keyringController;
+    this.coreKeyringController = opts.coreKeyringController;
```

Comment:
As for the `EncryptionPublicKeyController`, perhaps would be better to leave the `MMIController` migration to a separate PR. Here there are a bit more direct interactions with `EthKeyringController` compared to `EncryptionPublicKeyController`:
- `addNewKeyring` - Available in KeyringController
- `getAccounts` - Available in KeyringController
- `getKeyringsByType` - Deprecated but available in KeyringController
- `getKeyringForAccount` - Deprecated but available in KeyringController
- `addNewAccount` - Available only for HD Keyring, but it looks like MMIController needs to call this method on keyring of other types, which is also not supported at the moment by KeyringController
- `persistAllKeyrings` - Not available on KeyringController and only needed when using Keyrings directly (which unfortunately is the case for MMI)

From what I can see after this fast analysis, looks like our main necessity here is a way to call `persistAllKeyrings` from the core KeyringController

Reply:
I'd strongly encourage that we move away from this pattern of passing in an entire controller anyway. This pattern has caused us a great deal of trouble in the past with the older controllers, as it made it difficult to audit which methods were being used where. Instead we've been passing in individual functions instead (e.g. pass in `addNewAccount` directly).